bitcash/format.py

```diff
@@ -35,10 +43,10 @@ def verify_sig(signature, data, public_key):
     return _vs(signature, data, public_key)
 
 
-def address_to_public_key_hash(address):
+def address_to_public_key_hash(address, regtest=False):
```

Comment:
Are regtest addresses obvious? Can cashaddress detect this?

Reply:
They should start with `bchreg:`, so yes we don't need a regtest flag

Reply:
Unfortunately no. Regtest addresses are not always automatically determinable. If they are in 'cash address' format, then they do have the 'bchreg' prefix, however currently there are cases where a legacy address is passed to this function.

Legacy addresses (P2PKH) start with '1' on mainnet, and 'm' or 'n' on testnet. The format for legacy regtest addresses is *identical* to the legacy testnet format. Take the following **legacy address** for example:

`mxu5gDLNZ4yhoPiDy5u728nF64WJWqz8Nh`

It begins with 'm' therefore it's a testnet/regtest formatted address.
Converting it to a **testnet cash address** would output:

```bash
$ cashaddress.convert.to_cash_address("mxu5gDLNZ4yhoPiDy5u728nF64WJWqz8Nh", regtest=False)
> bchtest:qzl2juarkxncwvmlu8v5wkcjwhv0nv76m5gvtf6ka6
```

and converting it to a **regtest cash address** would output:

```bash
$ cashaddress.convert.to_cash_address("mxu5gDLNZ4yhoPiDy5u728nF64WJWqz8Nh", regtest=True)
> bchreg:qzl2juarkxncwvmlu8v5wkcjwhv0nv76m5jsage97u
```

If you strip off the prefix and checksum, you can see that the two are identical.

```
bchreg:  | qzl2juarkxncwvmlu8v5wkcjwhv0nv76m5 | jsage97u
bchtest: | qzl2juarkxncwvmlu8v5wkcjwhv0nv76m5 | gvtf6ka6
```

Since this `address_to_public_key_hash` function uses the `to_cash_address` function of the cashaddress library, and the `to_cash_address` necessitates a regtest flag to distinguish between the two networks, it was necessary for me to add the 'regtest' flag in this function and the several others. Without it, Bitcash was trying to use 'bchtest' addresses when doing things with PrivateKeyRegtest objects.

Side note: It defaults to False for the sake of backwards compatibility. Only the PrivateKeyRegtest class specifically sets it to True to provide the expected outputs. Until legacy addresses are removed from Bitcash, I think having this flag is an unfortunate necessity. I'm open to other implementations, but the flag seemed like the most obvious way to do it.

Reply:
I think the idea would be to drop support for legacy addresses.

What are the cases where legacy is needed?

Reply:
I'm open to dropping legacy addresses altogether.

Reply:
@teran-mckinney  Very well then. [Legacy support has been removed](https://github.com/pybitcash/bitcash/pull/82/commits/e90ce7868f4a091759cb3f8970274c6771a27fec) (along with this particular 'regtest' flag).